### PR TITLE
Do not set the server_version

### LINKED ISSUE #1696: Contao v4.9 doesn't work with MySQL 8 by default
**Affected version(s)**
Contao v4.9.2

**Stack used**
- MySQL v8.0.20 (Linux on x86_64)
- PHP-FPM v7.4
- Apache Server v2
- Docker 20.10.8

**Composer file**

```json
{
    "name": "contao/managed-edition",
    "type": "project",
    "description": "Contao Open Source CMS",
    "license": "LGPL-3.0-or-later",
    "authors": [
        {
            "name": "Leo Feyer",
            "homepage": "https://github.com/leofeyer"
        }
    ],
    "require": {
        "php": ">=7.2",
        "contao/calendar-bundle": "^4.9",
        "contao/comments-bundle": "^4.9",
        "contao/conflicts": "@dev",
        "contao/faq-bundle": "^4.9",
        "contao/listing-bundle": "^4.9",
        "contao/manager-bundle": "4.9.*",
        "contao/news-bundle": "^4.9",
        "contao/newsletter-bundle": "^4.9"
    },
    "conflict": {
        "contao-components/installer": "<1.3"
    },
    "extra": {
        "contao-component-dir": "assets",
        "symfony": {
            "require": "^4.4"
        }
    },
    "scripts": {
        "post-install-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ],
        "post-update-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ]
    }
}
```

**Description**

During the testing of a fresh Contao installation we got few MySQL errors and only the first three tables were created:

```
tl_crawl_queue
tl_cron_job
tl_remember_me
tl_trusted_device
```

**Complete error log trace**

`
[2020-05-04 00:00:00] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\SyntaxErrorException: "An exception occurred while executing 'CREATE TABLE tl_article (id INT UNSIGNED AUTO_INCREMENT NOT NULL, pid INT UNSIGNED DEFAULT 0 NOT NULL, sorting INT UNSIGNED DEFAULT 0 NOT NULL, tstamp INT UNSIGNED DEFAULT 0 NOT NULL, title VARCHAR(255) DEFAULT '' NOT NULL, alias VARCHAR(255) DEFAULT '' NOT NULL COLLATE `utf8mb4_bin`, author INT UNSIGNED DEFAULT 0 NOT NULL, inColumn VARCHAR(32) DEFAULT 'main' NOT NULL, keywords TEXT DEFAULT NULL, showTeaser CHAR(1) DEFAULT '' NOT NULL, teaserCssID VARCHAR(255) DEFAULT '' NOT NULL, teaser TEXT DEFAULT NULL, printable VARCHAR(255) DEFAULT '' NOT NULL, customTpl VARCHAR(64) DEFAULT '' NOT NULL, protected CHAR(1) DEFAULT '' NOT NULL, groups BLOB DEFAULT NULL, guests CHAR(1) DEFAULT '' NOT NULL, cssID VARCHAR(255) DEFAULT '' NOT NULL, published CHAR(1) DEFAULT '' NOT NULL, start VARCHAR(10) DEFAULT '' NOT NULL, stop VARCHAR(10) DEFAULT '' NOT NULL, INDEX alias (alias), INDEX pid_start_stop_published_sorting (pid, start, stop, published, sorting), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB ROW_FORMAT = DYNAMIC':  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups BLOB DEFAULT NULL, guests CHAR(1) DEFAULT '' NOT NULL, cssID VARCHAR(255)' at line 1" at /var/www/share/project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 79 {"exception":"[object] (Doctrine\\DBAL\\Exception\\SyntaxErrorException(code: 0): An exception occurred while executing 'CREATE TABLE tl_article (id INT UNSIGNED AUTO_INCREMENT NOT NULL, pid INT UNSIGNED DEFAULT 0 NOT NULL, sorting INT UNSIGNED DEFAULT 0 NOT NULL, tstamp INT UNSIGNED DEFAULT 0 NOT NULL, title VARCHAR(255) DEFAULT '' NOT NULL, alias VARCHAR(255) DEFAULT '' NOT NULL COLLATE `utf8mb4_bin`, author INT UNSIGNED DEFAULT 0 NOT NULL, inColumn VARCHAR(32) DEFAULT 'main' NOT NULL, keywords TEXT DEFAULT NULL, showTeaser CHAR(1) DEFAULT '' NOT NULL, teaserCssID VARCHAR(255) DEFAULT '' NOT NULL, teaser TEXT DEFAULT NULL, printable VARCHAR(255) DEFAULT '' NOT NULL, customTpl VARCHAR(64) DEFAULT '' NOT NULL, protected CHAR(1) DEFAULT '' NOT NULL, groups BLOB DEFAULT NULL, guests CHAR(1) DEFAULT '' NOT NULL, cssID VARCHAR(255) DEFAULT '' NOT NULL, published CHAR(1) DEFAULT '' NOT NULL, start VARCHAR(10) DEFAULT '' NOT NULL, stop VARCHAR(10) DEFAULT '' NOT NULL, INDEX alias (alias), INDEX pid_start_stop_published_sorting (pid, start, stop, published, sorting), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB ROW_FORMAT = DYNAMIC':\n\nSQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups BLOB DEFAULT NULL, guests CHAR(1) DEFAULT '' NOT NULL, cssID VARCHAR(255)' at line 1 at /var/www/share/project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:79, Doctrine\\DBAL\\Driver\\PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups BLOB DEFAULT NULL, guests CHAR(1) DEFAULT '' NOT NULL, cssID VARCHAR(255)' at line 1 at /var/www/share/project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:83, PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups BLOB DEFAULT NULL, guests CHAR(1) DEFAULT '' NOT NULL, cssID VARCHAR(255)' at line 1 at /var/www/share/project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:78)"} []
`

**Environment variables used**
- APP_ENV: `prod`
- SECRET: `abcdef1230`
- TRUSTED_PROXIES: `127.0.0.1,REMOTE_ADDR`
- DATABASE_URL: `mysql://username:passwd@database:3306/dbname`

**Steps to reproduce the issue**

https://github.com/contao/contao/issues/1696#issuecomment-944373619

**Screenshots**

https://github.com/contao/contao/issues/1696#issuecomment-945426969

### LINKED ISSUE #3345: Find a better way to set the doctrine server_version
**Affected version(s)**

Contao 4.9

**Description**

I want to find a better solution for https://github.com/contao/contao/blob/5fdcb347db6e2f18e999ee5e673c7d0c916bcee9/manager-bundle/src/ContaoManager/Plugin.php#L311-L328 as this currently also gets triggered for Contao (outside of the install-tool) and therefore can cause issues like in https://github.com/contao/contao/issues/3305#issuecomment-899730840

Related: https://github.com/contao/core-bundle/issues/1422 https://github.com/contao/installation-bundle/issues/54 #1696

### PR DESCRIPTION
**Pull request by ausi**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #1696

For the record as described in https://github.com/contao/contao/pull/3623#issuecomment-952708988 the `CommandSchedulerListener` must only load the `Cron` service when it needs it. Otherwise the install tool would fail when terminating because the `Cron` service needs ORM metadata which itself needs to know the `server_version`.

### CODE REVIEW

**Review by m-vo — DISMISSED** [review]
Awesomesauce!

**Comment by fritzmg** [comment]
Very nice! What is the reason behind the `CommandSchedulerListener` changes though?

**Comment by m-vo** [comment]
The class gets instantiated in the kernel terminate event. And because it gets an entity repository injected, this will trigger warming the doctrine proxy classes again...

**Review by bytehead — DISMISSED** [review]

File: core-bundle/src/Resources/config/services.yml [hunk 1]
```diff
@@ -231,6 +231,12 @@ services:
         tags:
             - { name: data_collector, template: '@ContaoCore/Collector/contao.html.twig', id: contao }
 
+    Contao\CoreBundle\Doctrine\ORM\FailTolerantProxyCacheWarmer:
```
Comment (Toflar) [inline core-bundle/src/Resources/config/services.yml:234]: This should not happen, if `server_version` was explicitly configured.

**Review by Toflar — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/config/services.yml (hunk 1 above)
Comment (m-vo) [reply core-bundle/src/Resources/config/services.yml:234]: Yes, we noted this as well, needs to be adjusted. :upside_down_face:

Comment (ausi) [reply core-bundle/src/Resources/config/services.yml:234]: Done in 2236de4a5ef3d8a1199fa593cf85f811e6a340c0

**Review by Toflar — APPROVED** [review]

**Review by m-vo — APPROVED** [review]

**Review by aschempp — CHANGES_REQUESTED** [review]
great catch! How about making the `Cron` service `lazy: true` instead? That would solve the problem for anyone using the cron service, not just our command scheduler.

**Comment by ausi** [comment]
I think using a service subscriber is more straight forward and doesn’t require to generate proxy classes that are not needed in 99 % of the cases.

**Review by fritzmg — APPROVED** [review]

**Comment by fritzmg** [comment]
Would it make sense to rewrite the Cron service to not use entities? Or would that not actually help in this case anyway?

**Comment by ausi** [comment]
> Would it make sense to rewrite the Cron service to not use entities?

IMO no. As we want to get rid of the install tool, this fix should be sufficient for now.

**Comment by aschempp** [comment]
> I think using a service subscriber is more straight forward and doesn’t require to generate proxy classes that are not needed in 99 % of the cases.

Dunno... its all the changes in https://github.com/contao/contao/pull/3623/files?diff=unified&w=0#diff-301de337aef887f559d97bb8997b22086044dc3c2621b2a35e1a801802016e3b and its test vs. one `lazy: true`...  But yeah both works 🤷

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**